geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/geotools/MetadataBackedDataStore.scala

```diff
@@ -407,5 +433,14 @@ object MetadataBackedDataStore {
   import org.locationtech.geomesa.utils.geotools.SimpleFeatureTypes.Configs._
 
   private val UnmodifiableUserDataKeys =
-    Set(TableSharing, TableSharingPrefix, IndexVisibilityLevel, IndexZ3Interval, IndexXzPrecision)
+    Set(
+      TableSharing,
+      TableSharingPrefix,
+      IndexVisibilityLevel,
+      IndexZ3Interval,
+      IndexXzPrecision,
+      IndexZShards,
+      IndexIdShards,
+      IndexAttributeShards
```

Comment:
Should one be able to change:
1. DefaultDtgField?
2. FidsAreUuidEncoded?
3. S3_INTERVAL_KEY?
4. TableCompressionType?
5. TableLogicalTime?
6. TableSharing?

Reply:
1. dtg is checked here (it can be changed if the field is renamed): https://github.com/locationtech/geomesa/pull/2615/files#diff-67b90d2402343d0802fec2ee9a973023R378
2. this one is ok to update
3. no, i'll add that one
3. that should be ok... it's only used in create schema and it's transparent to us
4. also should be ok
5. in the list already